pkg/rewards/claim.go

```diff
@@ -0,0 +1,312 @@
+		tokenLeaves = append(tokenLeaves, rewardscoordinator.IRewardsCoordinatorTokenTreeMerkleLeaf{
+			Token:              claimTokenLeaf.Token,
+			CumulativeEarnings: claimTokenLeaf.CumulativeEarnings,
+		})
+	}
+	return tokenLeaves
+
+}
+func readAndValidateClaimConfig(cCtx *cli.Context, logger logging.Logger) (*ClaimConfig, error) {
+	network := cCtx.String(flags.NetworkFlag.Name)
+	rpcUrl := cCtx.String(flags.ETHRpcUrlFlag.Name)
+	earnerAddress := gethcommon.HexToAddress(cCtx.String(flags.EarnerAddressFlag.Name))
+	output := cCtx.String(flags.OutputFileFlag.Name)
+	broadcast := cCtx.Bool(flags.BroadcastFlag.Name)
+	tokenAddresses := cCtx.String(TokenAddressesFlag.Name)
+	tokenAddressArray := stringToAddressArray(strings.Split(tokenAddresses, ","))
+	rewardsCoordinatorAddress := cCtx.String(RewardsCoordinatorAddressFlag.Name)
+	var err error
+	if rewardsCoordinatorAddress == "" {
+		rewardsCoordinatorAddress, err = utils.GetRewardCoordinatorAddress(utils.NetworkNameToChainId(network))
+		if err != nil {
+			return nil, err
+		}
+	}
+	logger.Debugf("Using Rewards Coordinator address: %s", rewardsCoordinatorAddress)
+
+	claimTimestamp := cCtx.String(ClaimTimestampFlag.Name)
+	if claimTimestamp != LatestClaimTimestamp {
+		return nil, errors.New("claim-timestamp must be 'latest'")
+	}
+
+	recipientAddress := gethcommon.HexToAddress(cCtx.String(RecipientAddressFlag.Name))
+	if recipientAddress == utils.ZeroAddress {
+		logger.Infof(
+			"Recipient address not provided, using earner address (%s) as recipient address",
+			earnerAddress.String(),
+		)
+		recipientAddress = earnerAddress
+	}
+	chainID := utils.NetworkNameToChainId(network)
+	logger.Debugf("Using chain ID: %s", chainID.String())
+
+	proofStoreBaseURL := cCtx.String(ProofStoreBaseURLFlag.Name)
+
+	// If empty get from utils
+	if proofStoreBaseURL == "" {
+		proofStoreBaseURL = utils.GetProofStoreBaseURL(network)
+
+		// If still empty return error
+		if proofStoreBaseURL == "" {
+			return nil, errors.New("proof store base URL not provided")
+		}
+	}
+	logger.Debugf("Using Proof store base URL: %s", proofStoreBaseURL)
+
+	env, network := getEnvAndNetwork(network)
+	logger.Debugf("Environment: %s, Network: %s", env, network)
+
+	// Get SignerConfig
+	signerConfig, err := common.GetSignerConfig(cCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ClaimConfig{
+		Network:                   network,
+		RPCUrl:                    rpcUrl,
+		EarnerAddress:             earnerAddress,
+		Output:                    output,
+		Broadcast:                 broadcast,
+		TokenAddresses:            tokenAddressArray,
+		RewardsCoordinatorAddress: gethcommon.HexToAddress(rewardsCoordinatorAddress),
+		ChainID:                   chainID,
+		ProofStoreBaseURL:         proofStoreBaseURL,
+		Environment:               env,
+		RecipientAddress:          recipientAddress,
+		SignerConfig:              *signerConfig,
+	}, nil
+}
+
+func getEnvAndNetwork(network string) (string, string) {
```

Comment:
Im guessing this is taken from the rewards-updater, which is kinda out of date.

This is the correct list of environments: https://github.com/Layr-Labs/proofs-service/blob/ab56d567f5ec759103e2a95488d278431729e75e/internal/config/config.go#L35-L46

Reply:
yea I am going to clean this up a bit. env and network is confusing

Reply:
clear now 0aeb8f1a8894dac95c5306f67f45187b8d4b0e95